libs/langchain-azure-openai/src/chat_models.ts

```diff
@@ -0,0 +1,687 @@
+    ];
+  }
+
+  get lc_secrets(): { [key: string]: string } | undefined {
+    return {
+      azureOpenAIApiKey: "AZURE_OPENAI_API_KEY",
+      azureOpenAIEndpoint: "AZURE_OPENAI_API_ENDPOINT",
+      azureOpenAIApiDeploymentName: "AZURE_OPENAI_API_DEPLOYMENT_NAME",
+    };
+  }
+
+  get lc_aliases(): Record<string, string> {
+    return {
+      modelName: "model",
+      azureOpenAIApiKey: "azure_openai_api_key",
+      azureOpenAIEndpoint: "azure_openai_api_endpoint",
+      azureOpenAIApiDeploymentName: "azure_openai_api_deployment_name",
+    };
+  }
+
+  lc_serializable = true;
+
+  functions?: FunctionDefinition[] | undefined;
+
+  functionCall?: FunctionCallPreset | FunctionName | undefined;
+
+  azureExtensionOptions?: AzureExtensionsOptions | undefined;
+
+  maxTokens?: number | undefined;
+
+  temperature: number;
+
+  topP: number;
+
+  logitBias?: Record<string, number> | undefined;
+
+  user?: string | undefined;
+
+  n: number;
+
+  presencePenalty: number;
+
+  frequencyPenalty: number;
+
+  stop?: string[] | undefined;
+
+  streaming: boolean;
+
+  modelName: string;
+
+  modelKwargs?: OpenAIChatInput["modelKwargs"];
+
+  timeout?: number | undefined;
+
+  azureOpenAIEndpoint?: string;
+
+  azureOpenAIApiKey?: string;
+
+  azureOpenAIApiCompletionsDeploymentName?: string;
+
+  private client: AzureOpenAIClient;
+
+  constructor(
+    fields?: Partial<OpenAIChatInput> &
+      Partial<AzureOpenAIInput> &
+      BaseChatModelParams & {
+        configuration?: AzureOpenAIClientOptions;
+      }
+  ) {
+    super(fields ?? {});
+
+    this.azureOpenAIEndpoint =
+      fields?.azureOpenAIEndpoint ??
+      getEnvironmentVariable("AZURE_OPENAI_API_ENDPOINT");
+
+    this.azureOpenAIApiCompletionsDeploymentName =
+      fields?.azureOpenAIEmbeddingsApiDeploymentName ??
+      getEnvironmentVariable("AZURE_OPENAI_API_DEPLOYMENT_NAME");
+
+    this.azureOpenAIApiKey =
+      fields?.azureOpenAIApiKey ??
+      getEnvironmentVariable("AZURE_OPENAI_API_KEY");
+
+    if (!this.azureOpenAIApiKey) {
+      throw new Error("Azure OpenAI API key not found");
+    }
+
+    if (!this.azureOpenAIEndpoint) {
+      throw new Error("Azure OpenAI Endpoint not found");
+    }
+
+    if (!this.azureOpenAIApiCompletionsDeploymentName) {
+      throw new Error("Azure OpenAI Completion Deployment name not found");
+    }
+
+    this.modelName = fields?.modelName ?? this.modelName;
+    this.modelKwargs = fields?.modelKwargs ?? {};
+    this.timeout = fields?.timeout;
+    this.temperature = fields?.temperature ?? this.temperature;
+    this.topP = fields?.topP ?? this.topP;
+    this.frequencyPenalty = fields?.frequencyPenalty ?? this.frequencyPenalty;
+    this.presencePenalty = fields?.presencePenalty ?? this.presencePenalty;
+    this.maxTokens = fields?.maxTokens;
+    this.n = fields?.n ?? this.n;
+    this.logitBias = fields?.logitBias;
+    this.stop = fields?.stop;
+    this.user = fields?.user;
+    this.functions = fields?.functions;
```

Comment:
We've been binding these at runtime like this:

https://js.langchain.com/docs/modules/model_io/chat/function_calling

Would prefer to keep it that way and standardize around the raw `function_call` to avoid confusion.

Reply:
Per the comments, this has been removed and tested.